### DISCUSSION TITLE
[Unstable] ice fragments have collision now

### DISCUSSION THREAD

**Discussion by LePiromano** [FakeFishGames/Barotrauma#15069] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

don't know how you pulled it off but the fragments created by ice spires/cave obstructions/floating ice bergs now have colission and will cause heavy damage if they fall on/if rammed into by submarine, players and projectiles still don't collide with them

this actually works really smoothly and i see it being a really cool new feature if you tweak the damage of the fragments and/or make them destructible

### Reproduction steps

break a spire on top of the sub/break a spire and drive into the fragments

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.7.0.1 (Unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by girlyguppy** [comment]
reminder to not remove this, just balance it 🙏 

thoughts:
-allow it to crush enemies/creatures against the sub
-allow the fragments to be shot into harmless small pieces 
-have spalling when shooting the ice or the fragments so that shooting right next to an entity can harm them
-have the fragments weigh down the submarine until destroyed

**Reply by NotWendy** [reply]
Agreed, this bug would go hard as a feature, if done right.

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/commit/e09081f0fdf4f17cfbd2f4dd84b01115ed8f7322 (commit in our private repo).

Unfortunately we went with just making them go through subs again: even though I agree it's pretty cool to have them collide with subs, it would be non-trivial to get them to work well (with the network syncing, proper collision handling with subs and characters, AIs being able to react to the fragments...) and in my opinion it doesn't feel worth it for such a minor addition (at least for the time being).

**Reply by girlyguppy** [reply]
AUGH

level hazards will never be real 😔
